File: ios/Classes/Data/SegmentationData.swift
//
//  SegmentationData.swift
//  exponea
//

import Foundation
import ExponeaSDK

struct SegmentationData {
    public let instanceId: String
    public let data: [SegmentDTO]

    func toMap() -> [String: Any] {
        return [
            "instanceId": instanceId,
            "data": data.map { return ["id" : $0.id, "segmentation_id" : $0.segmentationId] }
        ]
    }
}

File: android/src/main/kotlin/com/exponea/data/SegmentationData.kt
package com.exponea.data

import com.exponea.sdk.models.Segment

data class SegmentationData(
    val instanceId: String,
    val data: List<Segment>,
) {
    fun toMap(): Map<String, Any?> {
        return mapOf(
            "instanceId" to instanceId,
            "data" to data,
        )
    }
}
